extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -29,85 +75,74 @@ protected function setGlobalDisplayVars() {
 		$this->wg->SuppressWikiHeader = true;
 		$this->wg->SuppressPageHeader = true;
 		$this->wg->SuppressFooter = true;
+		$this->wg->Out->setPageTitle( 'Image Review tool' );
+		$this->wg->Out->enableClientCache( false );
+	}
 
-		$this->wg->Out->setPageTitle($this->getPageTitle());
+	private function setCache() {
+		$this->response->setCacheValidity( WikiaResponse::CACHE_DISABLED );
+		$this->response->sendHeaders();
+	}
 
-		$this->wg->Out->enableClientCache( false );
+	private function setAssets() {
+		$this->response->addAsset('extensions/wikia/ImageReview/js/jquery.onImagesLoad.js');
+		$this->response->addAsset('extensions/wikia/ImageReview/js/ImageReview.js');
+		$this->response->addAsset('extensions/wikia/ImageReview/css/ImageReview.scss');
 	}
 
 	public function index() {
-		$this->setGlobalDisplayVars();
-
 		$this->action = $this->parseAction();
-		$this->order = $this->getOrderingMethod();
-		$this->ts = $this->request->getVal( 'ts' );
+		$this->order = $this->request->getInt( 'sort' );
+		$this->timestamp = $this->request->getVal( 'ts' );
 
 		$this->checkUserPermissions();
-		$this->checkRedirect();
 
-		$this->setCache();
-		$this->setAssets();
-
-		$this->helper = $this->getHelper();
-
-		if( $this->wg->request->wasPosted() ) {
-			$this->handlePostAction();
+		if ( $this->checkHandleStats() ) {
+			// Return true to stop executing index but make sure to render the template
+			return true;
 		}
 
-		$user_key = $this->helper->getUserTsKey();
-		$newestTs = $this->wg->Memc->get( $user_key );
-
-		if ( $this->ts > $newestTs ) {
-			WikiaLogger::instance()->info( 'ImageReviewLog', [
-				'method' => __METHOD__,
-				'message' => "I've got the newest ts ({$this->ts}), I won't refetch the images",
-			]);
-			$this->imageList = array();
-			$this->wg->memc->set( $user_key, $this->ts, 3600 /* 1h */ );
-		} else {
-			$this->imageList = $this->helper->refetchImageListByTimestamp( $this->ts );
+		if ( $this->wg->request->wasPosted() ) {
+			$this->handlePostAction();
 		}
 
-		/* SUS-541 / Mix <mix@wikia.com> / scope: the following if block */
-		if ( count( $this->imageList ) < ImageReviewHelper::LIMIT_IMAGES ) {
-			$this->logImageListCompleteness( 'warning' );
+		if ( $this->checkRedirect() ) {
+			// Return false to stop any template rendering
+			return false;
 		}
 
-		if ( count( $this->imageList ) == 0 ) {
```

Comment:
All this crazy logic has been moved into the `ImageListGetter` and `ImageCountGetter` classes and simplified.